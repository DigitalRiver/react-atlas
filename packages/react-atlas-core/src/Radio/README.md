###### Default radio:

    <RadioGroup name="test">
		<Radio label="Option 1" value="first"/>
		<Radio label="Option 2" value="second"/>
	</RadioGroup>

###### Check radio by default:

    <RadioGroup name="test">
		<Radio label="Option 1" value="first"/>
		<Radio label="Option 2" value="second" defaultChecked/>
	</RadioGroup>

###### Disabled radio:

    <RadioGroup name="test">
		<Radio label="Option 1" value="first"/>
		<Radio label="Option 2" value="second" disabled/>
		<Radio label="Option 3" value="third"/>
	</RadioGroup>

###### Hidden radio:

    <RadioGroup name="test">
		<Radio label="Option 1" value="first"/>
		<Radio label="Option 2" value="second" hidden/>
		<Radio label="Option 3" value="third"/>
	</RadioGroup>

###### Radio with left-positioned label:

    <RadioGroup name="test">
		<Radio label="Option 1" labelPosition="left" value="first"/>
		<Radio label="Option 2" value="second"/>
	</RadioGroup>

###### Custom onClick method:

    <RadioGroup name="test">
		<Radio label="Custom onClick" value="first" onClick={ () => { alert('clicked!') } }/>
		<Radio label="Normal Radio" value="second"/>
	</RadioGroup>

###### Custom function before change:

    <RadioGroup name="test">
		<Radio 
			label="Custom onBeforeChange"
			value="first" 
			onBeforeChange={ function(value){ if (!value) { let accept = confirm("Do you want to check this?"); return accept; } return true; } }/>
		<Radio label="Normal Radio" value="second"/>
	</RadioGroup>
