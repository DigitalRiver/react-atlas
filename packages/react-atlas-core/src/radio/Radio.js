import React, { Component, PropTypes } from 'react';
import { classNames } from '../utils';
import themeable from 'react-themeable';

const Radio = ({className, disabled, inline, name, label, ...props}) => {
  const theme = themeable(props.theme);
  const labelClasses = classNames({
    block: !inline,
    inline,
    disabled
  }, className);

  return (
    <label {...theme(1, ...labelClasses)}>
      <input
        {...props}
        disabled={disabled}
        name={name}
        type="radio"
      />
      {label ? <span> {label}</span> : null}
    </label>
  );
};

Radio.propTypes = {
  defaultChecked: PropTypes.bool,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  inline: PropTypes.bool,
  label: PropTypes.string,
  name: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onClick: PropTypes.func,
  onFocus: PropTypes.func,
  value: PropTypes.any
};

Radio.defaultProps = {
  className: '',
  disabled: false,
  inline: false
};

Radio.styleguide = {
  category: 'Form Components',
  index: '3.8',
  example: `
<section>
  <h5>Individual Radio Buttons</h5>
  <p>Not aware of each other, use RadioGroup for that.</p>
  <Radio label="Checked Radio" value="checkedRadio" defaultChecked />
  <Radio label="Disabled Radio" value="disabledRadio" disabled/>
  <Radio label="Inlined 1" value="inlined1" className="someClassIMadeUp" inline />
  <Radio label="Inlined 2" value="inlined2" inline />
</section>
`
};

export default Radio;
