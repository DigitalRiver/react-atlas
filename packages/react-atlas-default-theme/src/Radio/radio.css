.radioLabel {
	composes: cursor-pointer from '../styles.css';
}

.radioInput {
	position: absolute;
	composes: opacity-0 from '../styles.css';
}

.radioInput + span { 
	position: relative;
}

	/* we can't use composes here due to pseudo-element */
	.radioInput + span:before {
		cursor: pointer;
		position: relative;
		content: "";
	    display: inline-block;
	    margin-right: 5px;
	    top: 5px;
	    height: 18px;
	    width: 18px;
		background: white;
		background: -webkit-linear-gradient(#ffffff, #e6e6e6);
		background: -o-linear-gradient(#ffffff, #e6e6e6);
		background: -moz-linear-gradient(#ffffff, #e6e6e6);
		background: linear-gradient(#ffffff, #e6e6e6);
		border: 1px solid #999999;
		border-radius: 50%;
		-webkit-border-radius: 50%;
		-moz-border-radius: 50%;
		-o-border-radius: 50%;
	}

	.radioInput:checked + span:before {
		background: #007eb5;
		box-shadow: inset 0px 0px 0px 3px white;
		-webkit-box-shadow: inset 0px 0px 0px 3px white;
		-moz-box-shadow: inset 0px 0px 0px 3px white;
		-o-box-shadow: inset 0px 0px 0px 3px white;
	}


.block {
	composes: marg-b-1 from '../styles.css';
	display: block;
	line-height: 30px;
}

.inline {
	composes: marg-r-2 from '../styles.css';
	display: inline-block;
}

.disabled {
	composes: gray from '../styles.css';
}