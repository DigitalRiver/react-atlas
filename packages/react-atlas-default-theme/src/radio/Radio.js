import { Radio } from "react-atlas-core";
import style from "./radio.css";
import createComponent from "../createComponent";

export default createComponent(Radio, style);
