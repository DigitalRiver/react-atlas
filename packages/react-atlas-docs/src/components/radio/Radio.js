import React, { PropTypes } from "react";
import * as Atlas from 'react-atlas';

export default class Radio extends React.Component {
  constructor(props){
    super(props)
  }
  render() {
    return (
      <Atlas.Radio {...this.props}></Atlas.Radio>
    )
  }
}