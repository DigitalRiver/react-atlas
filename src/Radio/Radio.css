@import '../styles/variables.css';

.disabled {
    composes: default-text from '../styles.css';
}

.relative {
    position: relative;
}

.padding {
    padding-bottom: 5px;
}

.input {
    composes: relative;
    top: 1px;
}

.radio {
    position: absolute;
    top: 0;
    left: 0;
    height: 16px;
    width: 16px;
    display: inline-block;
    composes: cursor-pointer from '../styles.css';
    composes: default-border from '../styles.css';
    composes: circle from '../styles.css';
    background: -webkit-linear-gradient(#fff, #e6e6e6); /* For Safari 5.1 to 6.0 */
    background: -o-linear-gradient(#fff, #e6e6e6); /* For Opera 11.1 to 12.0 */
    background: -moz-linear-gradient(#fff, #e6e6e6); /* For Firefox 3.6 to 15 */
    background: linear-gradient(#fff, #e6e6e6); /* Standard syntax */
}

.disabled {
    opacity: 0.65;
    composes: cursor-not-allowed from '../styles.css';
    pointer-events: none;
}

.hidden {
    display: none;
    visibility: hidden;
}

.checked {
    composes: radio;
    composes: white-inset-shadow from '../styles.css';
    background: var(--primary);
}

.focus {
    box-shadow: 0 0 6px rgba(35, 173, 255, 1);
}

.checkmark {
    composes: relative;
    top: 0px;
}

.error {
    border: solid 1px var(--error);
}

.label {
    composes: default-font from '../styles.css';
    cursor: pointer;
    white-space: nowrap;
    padding-right: 5px;
    margin-left: 5px;
}

.label_left {
    composes: label;
    padding-right: 5px;
    margin-left: 0;
    margin-right: 5px;
}

.inline_block {
    display: inline-block;
}

.radio_padding {
    padding: 7px 10px 0 0;
}

.float_left {
    float: left;
    padding-right: 3px;
}

.float_right {
    float: right;
    padding-left: 3px;
    position: relative;
}

.input_style {
    opacity: 0; /*this needs to stay here to avoid display issues in IE*/
}