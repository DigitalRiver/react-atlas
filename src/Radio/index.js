import Radio from "./Radio.js";
export { Radio };
