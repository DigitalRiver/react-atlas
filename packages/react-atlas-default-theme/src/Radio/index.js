import * as RadioStyle from "./radio.css";
export { RadioStyle };
