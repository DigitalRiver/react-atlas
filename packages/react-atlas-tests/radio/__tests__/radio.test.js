import React from "react";
import { mount } from "enzyme";
import { RadioCore } from "../../../react-atlas-core/src/Radio/index";

import renderer from "react-test-renderer";

describe("Test Radio component render", () => {
  it("Render correctly", () => {
    const tree = renderer
      .create(
        <RadioCore
          label={"Checked Radio"}
          value={"checkedRadio"}
          defaultChecked={true}
          className={"class"}
          title={"TITLE"}
          hidden={false}
          inline={true}
          name={"Mr radio"}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Testing radio component", () => {
  it("Radio - Basic test", function() {
    const result = mount(
      <RadioCore
        label={"Checked Radio"}
        value={"checkedRadio"}
        defaultChecked={true}
        className={"class"}
        title={"TITLE"}
        hidden={false}
        inline={true}
        name={"Mr radio"}
      />
    );
  });

  it("Radio - Basic test", function() {
    const result = mount(
      <RadioCore
        label="Checked Radio"
        value="checkedRadio"
        labelPosition={"left"}
      />
    );
  });

  it("Radio - Default checked test", function() {
    const rad = mount(
      <RadioCore
        checked={true}
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
      />
    );
    expect(rad.props().checked).toEqual(true);
  });

  it("Radio - Simple click test", function() {
    const rad = mount(
      <RadioCore
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
      />
    );
    rad.simulate("click");
  });

  it("Radio - Simple click test (disabled)", function() {
    const rad = mount(
      <RadioCore
        disabled={true}
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
      />
    );
    rad.simulate("click");
  });

  it("Radio - Simple click test (with onBeforeChange)", function() {
    const rad = mount(
      <RadioCore
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
        onClick={function() {
          console.log("onclick");
        }}
        onChange={function() {
          console.log("onchange");
        }}
        onBeforeChange={function() {
          console.log("onbeforechange");
        }}
      />
    );
    rad.simulate("click");
  });

  it("Radio - Simple click test (without onBeforeChange)", function() {
    const rad = mount(
      <RadioCore
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
        onClick={function() {
          console.log("onclick");
        }}
        onChange={function() {
          console.log("onchange");
        }}
      />
    );
    rad.simulate("click");
  });

  it("Radio - Simple click test (without onBeforeChange) checked", function() {
    const rad = mount(
      <RadioCore
        checked={true}
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
        onClick={function() {
          console.log("onclick");
        }}
        onChange={function() {
          console.log("onchange");
        }}
      />
    );
    rad.simulate("click");
  });
});
