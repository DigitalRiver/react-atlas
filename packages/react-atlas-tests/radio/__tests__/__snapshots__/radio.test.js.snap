// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Radio component render Render correctly 1`] = `
<div
  onClick={[Function]}
  style={undefined}
  styleName="inline_block radio_padding"
>
  <div
    styleName="inline_block relative padding"
  >
    <label
      className="class"
      styleName="label"
      title="TITLE"
    >
      Checked Radio
    </label>
    <div
      styleName="float_left"
    >
      <div
        styleName="container"
      >
        <input
          className="ra_Input__max ra_Input__opacity"
          defaultChecked={true}
          inline={true}
          label="Checked Radio"
          name="Mr radio"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          styleName=""
          title="TITLE"
          type="radio"
          value="checkedRadio"
        />
      </div>
      <div
        styleName="not_checked"
      />
    </div>
  </div>
</div>
`;
